catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,367 @@
+                                            .getId(),
+                                    OVERVIEW,
+                                    renderedImage,
+                                    metacard,
+                                    calculateOverviewWidth(renderedImage),
+                                    calculateOverviewHeight(renderedImage));
+
+                            T overviewProcessResourceItem = createResourceItem(processResourceItem,
+                                    overviewImageProcessResource);
+                            newProcessResourceItems.add(overviewProcessResourceItem);
+                        } catch (IOException e) {
+                            LOGGER.debug("Unable to generate Overview image from nitf. {}.",
+                                    e.getMessage());
+                        }
+
+                        // original
+                        try {
+                            ProcessResource originalImageProcessResource = createDerivedImage(
+                                    processResourceItem.getMetacard()
+                                            .getId(),
+                                    ORIGINAL,
+                                    renderedImage,
+                                    metacard,
+                                    renderedImage.getWidth(),
+                                    renderedImage.getHeight());
+
+                            T originalProcessResourceItem = createResourceItem(processResourceItem,
+                                    originalImageProcessResource);
+                            newProcessResourceItems.add(originalProcessResourceItem);
+                        } catch (IOException e) {
+                            LOGGER.debug("Unable to generate Original image from nitf. {}.",
+                                    e.getMessage());
+                        }
+                    }
+                } catch (IOException | ParseException | NitfFormatException e) {
+                    LOGGER.debug("Unable to render image from nitf. {}.", e.getMessage());
+                }
+            }
+        }
+
+        processResourceItems.addAll(newProcessResourceItems);
+    }
+
+    private static <T extends ProcessResourceItem> T createResourceItem(T processResourceItem,
+            ProcessResource processResource) throws PluginExecutionException {
+        T newProcessResourceItem;
+        if (processResourceItem instanceof ProcessCreateItem) {
+            newProcessResourceItem = (T) new ProcessCreateItemImpl(processResource,
+                    processResourceItem.getMetacard());
+        } else if (processResourceItem instanceof ProcessUpdateItem) {
+            newProcessResourceItem = (T) new ProcessUpdateItemImpl(processResource,
+                    processResourceItem.getMetacard(),
+                    ((ProcessUpdateItem) processResourceItem).getOldMetacard());
+        } else {
```

Comment:
It would end up needing to have a method for each constructor of the `ProcessResourceItem`s implementers. For instance, one for `ProcessCreateItem(ProcessResource, Metacard)` and `ProcessUpdateItem(ProcessResource, Metacard, Metacard)`.